generator/ProtocolTestsGenerator/smithy-dotnet-codegen/src/main/java/software/amazon/smithy/dotnet/codegen/HttpProtocolTestGenerator.java

```diff
@@ -0,0 +1,510 @@
+                        generateErrorResponseTests(operation, operationIndex);
+                    });
+                });
+                context.writerDelegator().flushWriters();
+            });
+        }
+    }
+
+    private void generateErrorResponseTests(OperationShape operation, OperationIndex index) {
+        for(StructureShape error : index.getErrors(operation, service)){
+            error.getTrait(HttpResponseTestsTrait.class).ifPresent( trait ->{
+                for(HttpResponseTestCase httpResponseTestCase : trait.getTestCasesFor(AppliesTo.CLIENT)){
+                    generateErrorResponseTest(operation, error, httpResponseTestCase);
+                }
+            });
+        }
+    }
+
+    private void generateErrorResponseTest(OperationShape operation, StructureShape error, HttpResponseTestCase httpResponseTestCase) {
+        if(httpResponseTestCase.getDocumentation().isPresent()){
+            writer.writeXmlDocs(httpResponseTestCase.getDocumentation().get());
+        }
+        writer.write("[TestMethod]");
+        writer.write("[TestCategory(\"ProtocolTest\")]");
+        writer.write("[TestCategory(\"ErrorTest\")]");
+        writer.write("[TestCategory(\"$L\")]",protocolNamespace);
+        writer.openBlock("public void $LErrorResponse()\n{","}",httpResponseTestCase.getId(), () ->{
+            generateErrorResponseTestBlock(operation,error,httpResponseTestCase);
+        });
+        writer.write("\n");
+    }
+
+    private void generateErrorResponseTestBlock(OperationShape operation, StructureShape error, HttpResponseTestCase httpResponseTestCase) {
+        var outputShape =  model.expectShape(operation.getOutputShape(), StructureShape.class);
+        var responseSymbol = operation.getId().getName() + "Response";
+        //Arrange
+        arrangeResponseTestBlock(httpResponseTestCase);
+        var errorSymbol = error.getId().getName() + "Exception";
+        writer.writeSingleLineComment("Act");
+        writer.write("var errorResponse = new $LUnmarshaller().UnmarshallException(context, new $L(\"\"), (HttpStatusCode)Enum.ToObject(typeof(HttpStatusCode), $L));",responseSymbol, errorSymbol, httpResponseTestCase.getCode());
+        writer.writeSingleLineComment("Assert");
+        // For now we will just assert that we get the right exception type. Since exceptions don't take a parameterless constructor
+        // there is no simple way to set the message without dramatic alterations to the ValueNodeVisitor. It is worth thinking about
+        // creating a new ValueNodeVisitor that works differently for exceptions at a later time.
+        writer.write("Assert.IsInstanceOfType(errorResponse, typeof($L));",errorSymbol);
+    }
+
+    private void arrangeResponseTestBlock(HttpResponseTestCase httpResponseTestCase){
+        writer.addImport(protocolNamespace,"System.Net");
+        writer.writeSingleLineComment("Arrange");
+        writer.write("var webResponseData = new WebResponseData();");
```

Comment:
I'm thinking about how to manage imports. I assume System.Net is imported for WebResponseData, but that is unclear reading this code. So it is difficult for a code writer/reviewer to know that all the needed imports are added. Is there are way to do something like `addImport("System.Net", "WebResponseData")` which will cause the `using System.Net` to be added?

Reply:
yup, I will make this change to make it clearer what the imports are for. I had the logic for this in DotnetImportContainer but I never populated it in the CSharpWriter.